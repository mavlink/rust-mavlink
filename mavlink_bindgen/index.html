<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="API documentation for the Rust `mavlink_bindgen` crate."><title>mavlink_bindgen - Rust</title><script>if(window.location.protocol!=="file:")document.head.insertAdjacentHTML("beforeend","SourceSerif4-Regular-6b053e98.ttf.woff2,FiraSans-Italic-81dc35de.woff2,FiraSans-Regular-0fe48ade.woff2,FiraSans-MediumItalic-ccf7e434.woff2,FiraSans-Medium-e1aa3f0a.woff2,SourceCodePro-Regular-8badfe75.ttf.woff2,SourceCodePro-Semibold-aa29a496.ttf.woff2".split(",").map(f=>`<link rel="preload" as="font" type="font/woff2"href="../static.files/${f}">`).join(""))</script><link rel="stylesheet" href="../static.files/normalize-9960930a.css"><link rel="stylesheet" href="../static.files/rustdoc-528d32ef.css"><meta name="rustdoc-vars" data-root-path="../" data-static-root-path="../static.files/" data-current-crate="mavlink_bindgen" data-themes="" data-resource-suffix="" data-rustdoc-version="1.91.0-nightly (6ba0ce409 2025-08-21)" data-channel="nightly" data-search-js="search-e4956047.js" data-stringdex-js="stringdex-0e748618.js" data-settings-js="settings-c38705f0.js" ><script src="../static.files/storage-1b37d467.js"></script><script defer src="../crates.js"></script><script defer src="../static.files/main-b6a35c53.js"></script><noscript><link rel="stylesheet" href="../static.files/noscript-32bb7600.css"></noscript><link rel="alternate icon" type="image/png" href="../static.files/favicon-32x32-6580c154.png"><link rel="icon" type="image/svg+xml" href="../static.files/favicon-044be391.svg"></head><body class="rustdoc mod crate"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><rustdoc-topbar><h2><a href="#">Crate mavlink_bindgen</a></h2></rustdoc-topbar><nav class="sidebar"><div class="sidebar-crate"><h2><a href="../mavlink_bindgen/index.html">mavlink_<wbr>bindgen</a><span class="version">0.15.0</span></h2></div><div class="sidebar-elems"><ul class="block"><li><a id="all-types" href="all.html">All Items</a></li></ul><section id="rustdoc-toc"><h3><a href="#reexports">Crate Items</a></h3><ul class="block"><li><a href="#reexports" title="Re-exports">Re-exports</a></li><li><a href="#modules" title="Modules">Modules</a></li><li><a href="#structs" title="Structs">Structs</a></li><li><a href="#enums" title="Enums">Enums</a></li><li><a href="#functions" title="Functions">Functions</a></li></ul></section><div id="rustdoc-modnav"></div></div></nav><div class="sidebar-resizer" title="Drag to resize sidebar"></div><main><div class="width-limiter"><section id="main-content" class="content"><div class="main-heading"><h1>Crate <span>mavlink_<wbr>bindgen</span>&nbsp;<button id="copy-path" title="Copy item path to clipboard">Copy item path</button></h1><rustdoc-toolbar></rustdoc-toolbar><span class="sub-heading"><a class="src" href="../src/mavlink_bindgen/lib.rs.html#1-210">Source</a> </span></div><h2 id="reexports" class="section-header">Re-exports<a href="#reexports" class="anchor">§</a></h2><dl class="item-table reexports"><dt id="reexport.BindGenError"><code>pub use crate::error::<a class="enum" href="error/enum.BindGenError.html" title="enum mavlink_bindgen::error::BindGenError">BindGenError</a>;</code></dt></dl><h2 id="modules" class="section-header">Modules<a href="#modules" class="anchor">§</a></h2><dl class="item-table"><dt><a class="mod" href="binder/index.html" title="mod mavlink_bindgen::binder">binder</a></dt><dt><a class="mod" href="error/index.html" title="mod mavlink_bindgen::error">error</a></dt><dt><a class="mod" href="parser/index.html" title="mod mavlink_bindgen::parser">parser</a></dt></dl><h2 id="structs" class="section-header">Structs<a href="#structs" class="anchor">§</a></h2><dl class="item-table"><dt><a class="struct" href="struct.GeneratedBinding.html" title="struct mavlink_bindgen::GeneratedBinding">Generated<wbr>Binding</a></dt><dt><a class="struct" href="struct.GeneratedBindings.html" title="struct mavlink_bindgen::GeneratedBindings">Generated<wbr>Bindings</a></dt></dl><h2 id="enums" class="section-header">Enums<a href="#enums" class="anchor">§</a></h2><dl class="item-table"><dt><a class="enum" href="enum.XmlDefinitions.html" title="enum mavlink_bindgen::XmlDefinitions">XmlDefinitions</a></dt><dd>Specifies the source(s) of MAVLink XML definition files used for generating
Rust MAVLink dialect bindings.</dd></dl><h2 id="functions" class="section-header">Functions<a href="#functions" class="anchor">§</a></h2><dl class="item-table"><dt><a class="fn" href="fn.emit_cargo_build_messages.html" title="fn mavlink_bindgen::emit_cargo_build_messages">emit_<wbr>cargo_<wbr>build_<wbr>messages</a></dt><dd>Prints definitions for cargo that describe which files the generated code depends on, indicating when it has to be regenerated.</dd><dt><a class="fn" href="fn.format_generated_code.html" title="fn mavlink_bindgen::format_generated_code">format_<wbr>generated_<wbr>code</a></dt><dd>Formats generated code using <code>rustfmt</code>.</dd><dt><a class="fn" href="fn.generate.html" title="fn mavlink_bindgen::generate">generate</a></dt><dd>Generate Rust MAVLink dialect binding for dialects present in the given <code>xml_definitions</code>
into <code>destination_dir</code>.</dd></dl></section></div></main></body></html>