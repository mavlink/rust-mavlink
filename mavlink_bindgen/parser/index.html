<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="API documentation for the Rust `parser` mod in crate `mavlink_bindgen`."><title>mavlink_bindgen::parser - Rust</title><script>if(window.location.protocol!=="file:")document.head.insertAdjacentHTML("beforeend","SourceSerif4-Regular-6b053e98.ttf.woff2,FiraSans-Italic-81dc35de.woff2,FiraSans-Regular-0fe48ade.woff2,FiraSans-MediumItalic-ccf7e434.woff2,FiraSans-Medium-e1aa3f0a.woff2,SourceCodePro-Regular-8badfe75.ttf.woff2,SourceCodePro-Semibold-aa29a496.ttf.woff2".split(",").map(f=>`<link rel="preload" as="font" type="font/woff2"href="../../static.files/${f}">`).join(""))</script><link rel="stylesheet" href="../../static.files/normalize-9960930a.css"><link rel="stylesheet" href="../../static.files/rustdoc-069232aa.css"><meta name="rustdoc-vars" data-root-path="../../" data-static-root-path="../../static.files/" data-current-crate="mavlink_bindgen" data-themes="" data-resource-suffix="" data-rustdoc-version="1.91.0-nightly (425a9c0a0 2025-08-17)" data-channel="nightly" data-search-js="search-b069538f.js" data-settings-js="settings-5514c975.js" ><script src="../../static.files/storage-68b7e25d.js"></script><script defer src="../sidebar-items.js"></script><script defer src="../../static.files/main-eebb9057.js"></script><noscript><link rel="stylesheet" href="../../static.files/noscript-32bb7600.css"></noscript><link rel="alternate icon" type="image/png" href="../../static.files/favicon-32x32-6580c154.png"><link rel="icon" type="image/svg+xml" href="../../static.files/favicon-044be391.svg"></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle" title="show sidebar"></button></nav><nav class="sidebar"><div class="sidebar-crate"><h2><a href="../../mavlink_bindgen/index.html">mavlink_<wbr>bindgen</a><span class="version">0.15.0</span></h2></div><div class="sidebar-elems"><section id="rustdoc-toc"><h2 class="location"><a href="#">Module parser</a></h2><h3><a href="#structs">Module Items</a></h3><ul class="block"><li><a href="#structs" title="Structs">Structs</a></li><li><a href="#enums" title="Enums">Enums</a></li><li><a href="#functions" title="Functions">Functions</a></li></ul></section><div id="rustdoc-modnav"><h2 class="in-crate"><a href="../index.html">In crate mavlink_<wbr>bindgen</a></h2></div></div></nav><div class="sidebar-resizer" title="Drag to resize sidebar"></div><main><div class="width-limiter"><rustdoc-search></rustdoc-search><section id="main-content" class="content"><div class="main-heading"><div class="rustdoc-breadcrumbs"><a href="../index.html">mavlink_bindgen</a></div><h1>Module <span>parser</span><button id="copy-path" title="Copy item path to clipboard">Copy item path</button></h1><rustdoc-toolbar></rustdoc-toolbar><span class="sub-heading"><a class="src" href="../../src/mavlink_bindgen/parser.rs.html#1-1675">Source</a> </span></div><h2 id="structs" class="section-header">Structs<a href="#structs" class="anchor">§</a></h2><dl class="item-table"><dt><a class="struct" href="struct.MavEnum.html" title="struct mavlink_bindgen::parser::MavEnum">MavEnum</a></dt><dt><a class="struct" href="struct.MavEnumEntry.html" title="struct mavlink_bindgen::parser::MavEnumEntry">MavEnum<wbr>Entry</a></dt><dt><a class="struct" href="struct.MavField.html" title="struct mavlink_bindgen::parser::MavField">MavField</a></dt><dt><a class="struct" href="struct.MavMessage.html" title="struct mavlink_bindgen::parser::MavMessage">MavMessage</a></dt><dt><a class="struct" href="struct.MavProfile.html" title="struct mavlink_bindgen::parser::MavProfile">MavProfile</a></dt></dl><h2 id="enums" class="section-header">Enums<a href="#enums" class="anchor">§</a></h2><dl class="item-table"><dt><a class="enum" href="enum.MavType.html" title="enum mavlink_bindgen::parser::MavType">MavType</a></dt><dt><a class="enum" href="enum.MavXmlElement.html" title="enum mavlink_bindgen::parser::MavXmlElement">MavXml<wbr>Element</a></dt></dl><h2 id="functions" class="section-header">Functions<a href="#functions" class="anchor">§</a></h2><dl class="item-table"><dt><a class="fn" href="fn.extra_crc.html" title="fn mavlink_bindgen::parser::extra_crc">extra_<wbr>crc</a></dt><dd>CRC operates over names of the message and names of its fields
Hence we have to preserve the original uppercase names delimited with an underscore
For field names, we replace “type” with “mavtype” to make it rust compatible (this is
needed for generating sensible rust code), but for calculating crc function we have to
use the original name “type”</dd><dt><a class="fn" href="fn.generate.html" title="fn mavlink_bindgen::parser::generate">generate</a></dt><dd>Generate protobuf represenation of mavlink message set
Generate rust representation of mavlink message set with appropriate conversion methods</dd><dt><a class="fn" href="fn.parse_profile.html" title="fn mavlink_bindgen::parser::parse_profile">parse_<wbr>profile</a></dt></dl></section></div></main></body></html>