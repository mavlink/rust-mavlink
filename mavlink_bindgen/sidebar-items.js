window.SIDEBAR_ITEMS = {"enum":["XmlDefinitions"],"fn":["emit_cargo_build_messages","format_generated_code","generate"],"mod":["binder","error","parser"],"struct":["GeneratedBinding","GeneratedBindings"]};