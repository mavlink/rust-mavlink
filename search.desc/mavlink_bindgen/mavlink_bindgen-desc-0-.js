searchState.loadedDescShard("mavlink_bindgen", 0, "Prints definitions for cargo that describe which files the …\nFormats generated code using <code>rustfmt</code>.\nReturns the argument unchanged.\nReturns the argument unchanged.\nGenerate Rust MAVLink dialect binding for dialects present …\nCalls <code>U::from(self)</code>.\nCalls <code>U::from(self)</code>.\nRepresents a failure to create a Rust file for the …\nRepresents a failure to read a MAVLink definition file.\nRepresents a failure to read the MAVLink definitions …\nRepresents a failure to read a directory entry in the …\nReturns the argument unchanged.\nCalls <code>U::from(self)</code>.\nCompare two MavTypes\nCRC operates over names of the message and names of its …\nReturns the argument unchanged.\nReturns the argument unchanged.\nReturns the argument unchanged.\nReturns the argument unchanged.\nReturns the argument unchanged.\nReturns the argument unchanged.\nReturns the argument unchanged.\nGenerate protobuf represenation of mavlink message set …\nCalls <code>U::from(self)</code>.\nCalls <code>U::from(self)</code>.\nCalls <code>U::from(self)</code>.\nCalls <code>U::from(self)</code>.\nCalls <code>U::from(self)</code>.\nCalls <code>U::from(self)</code>.\nCalls <code>U::from(self)</code>.\nIf contains Some, the string represents the primitive type …\nUsed for crc calculation\nReturn rust equivalent of the primitive type of a MavType. …\nEmit reader of a given type\nReturn rust equivalent of a given Mavtype Used for …\nEmit writer of a given type")