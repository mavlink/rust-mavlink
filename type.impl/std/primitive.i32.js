(function() {
    var type_impls = Object.fromEntries([["mio_serial",[]],["nix",[["<details class=\"toggle implementors-toggle\" open><summary><section id=\"impl-ErrnoSentinel-for-i32\" class=\"impl\"><a class=\"src rightside\" href=\"src/nix/errno.rs.html#151-155\">Source</a><a href=\"#impl-ErrnoSentinel-for-i32\" class=\"anchor\">ยง</a><h3 class=\"code-header\">impl <a class=\"trait\" href=\"nix/errno/trait.ErrnoSentinel.html\" title=\"trait nix::errno::ErrnoSentinel\">ErrnoSentinel</a> for <a class=\"primitive\" href=\"https://doc.rust-lang.org/nightly/std/primitive.i32.html\">i32</a></h3></section></summary><div class=\"impl-items\"><section id=\"method.sentinel\" class=\"method trait-impl\"><a class=\"src rightside\" href=\"src/nix/errno.rs.html#152-154\">Source</a><a href=\"#method.sentinel\" class=\"anchor\">ยง</a><h4 class=\"code-header\">fn <a href=\"nix/errno/trait.ErrnoSentinel.html#tymethod.sentinel\" class=\"fn\">sentinel</a>() -&gt; Self</h4></section></div></details>","ErrnoSentinel","nix::pty::SessionId"]]],["serde",[]],["serde_core",[]],["tokio",[]]]);
    if (window.register_type_impls) {
        window.register_type_impls(type_impls);
    } else {
        window.pending_type_impls = type_impls;
    }
})()
//{"start":55,"fragment_lengths":[17,966,13,18,13]}