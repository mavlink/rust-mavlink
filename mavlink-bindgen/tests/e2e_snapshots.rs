use std::fs;
use std::path::PathBuf;

use insta::{self, assert_snapshot, glob};
use mavlink_bindgen::{format_generated_code, generate, XmlDefinitions};
use tempfile::TempDir;

fn definitions_dir() -> PathBuf {
    PathBuf::from(env!("CARGO_MANIFEST_DIR"))
        .join("tests")
        .join("definitions")
}

fn run_snapshot(def_file: &str) {
    let defs = definitions_dir();
    let tmp = TempDir::new().expect("tmp dir");
    let out_dir = tmp.path();

    let xml = defs.join(def_file);
    let result = generate(XmlDefinitions::Files(vec![xml]), out_dir).expect("generate ok");

    format_generated_code(&result);

    glob!(out_dir, "**/*.rs", |path| {
        let contents = fs::read_to_string(path).expect("read generated file");
        assert_snapshot!(def_file, contents);
    })
}

#[test]
fn snapshot_heartbeat() {
    run_snapshot("heartbeat.xml");
}

#[test]
fn snapshot_parameters() {
    run_snapshot("parameters.xml");
}

#[test]
fn snapshot_deprecated() {
    run_snapshot("deprecated.xml");
}
