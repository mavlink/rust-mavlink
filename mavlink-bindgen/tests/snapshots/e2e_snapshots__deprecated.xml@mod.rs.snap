---
source: mavlink-bindgen/tests/e2e_snapshots.rs
assertion_line: 26
expression: contents
---
#[allow(non_camel_case_types)]
#[allow(clippy::derive_partial_eq_without_eq)]
#[allow(clippy::field_reassign_with_default)]
#[allow(non_snake_case)]
#[allow(clippy::unnecessary_cast)]
#[allow(clippy::bad_bit_mask)]
#[allow(clippy::suspicious_else_formatting)]
#[cfg(feature = "deprecated")]
pub mod deprecated;
